addons/mail/models/ir_config_parameter.py

```diff
@@ -31,5 +31,6 @@ def set_param(self, key, value):
 
             elif value and group_mail_template_editor in group_user.implied_ids:
                 group_user.implied_ids -= group_mail_template_editor
+                self.env.ref('base.default_user').groups_id -= group_mail_template_editor
```

Comment:
shouldn't we re-add it if we re-check the the option ?

Reply:
as the group is implied on employees, it will be automatically added to new users anyway